pkg/gateway/fabric/geneve/internalnode_controller.go

```diff
@@ -65,10 +66,23 @@ func (r *InternalNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	klog.V(4).Infof("Reconciling internalnode %s", req.String())
 
-	if err := EnsureGeneveInterfacePresence(internalnode, r.Options); err != nil {
+	// The internal fabric has the same name of the gateway resource.
```

Comment:
It seems quite unsafe, can you at least define a variable internalNodeName with this comment, and then use that variable?

```suggestion
	// The internal fabric has the same name of the gateway resource.
    internalNodeName := r.Options.GwOptions.Name
    id, err := geneve.GetGeneveTunnelID(ctx, r.Client, internalNodeName, internalnode.Name)
```

Reply:
We could use the generic function `gateway.GenerateResourceName()`` that we used for other resources associated with the gateway (and also use it when creating the internalfabrics in the server_controller and client_controller)

Reply:
@fra98 I agree with you.